### ISSUE TITLE
stopped updating

### ISSUE THREAD

**Issue by jackshaw20** [bimmerconnected/bimmer_connected#265]
I have updated home assistant to latest release and everything was working fine up until yesterday.  I have 2 connected drive accounts and, they was both working fine but one has stopped updating as of yesterday.  this is the log;

Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/config_entries.py", line 236, in async_setup
result = await component.async_setup_entry(hass, self) # type: ignore
File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/__init__.py", line 137, in async_setup_entry
await _async_update_all()
File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/__init__.py", line 119, in _async_update_all
await hass.async_add_executor_job(_update_all)
File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
result = self.fn(*self.args, **self.kwargs)
File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/__init__.py", line 123, in _update_all
for entry in hass.data[DOMAIN][DATA_ENTRIES].values():
RuntimeError: dictionary changed size during iteration

**Comment by jackshaw20** [comment]
the account thats stopped working I can login on BMW connected drive site ok.

I have just restarted HA, this is the log:

Logger: homeassistant.config_entries
Source: components/bmw_connected_drive/__init__.py:123
First occurred: 21:59:32 (1 occurrences)
Last logged: 21:59:32

Error setting up entry steph@xxxxxxxxx.co.uk for bmw_connected_drive
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 236, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/__init__.py", line 137, in async_setup_entry
    await _async_update_all()
  File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/__init__.py", line 119, in _async_update_all
    await hass.async_add_executor_job(_update_all)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/__init__.py", line 123, in _update_all
    for entry in hass.data[DOMAIN][DATA_ENTRIES].values():
RuntimeError: dictionary changed size during iteration

**Comment by rikroe** [comment]
Hi, thanks for your issue! 

It looks like this is a race condition due to having two accounts: While the first account is set up and updates for the first time, the second account is added (therefore changing the dict's size).

What install type of HA are you using?
Would you be able to change **line 123** in `components/bmw_connected_drive/init.py` to the following:
```
for entry in hass.data[DOMAIN][DATA_ENTRIES].copy().values():
```

**Comment by jackshaw20** [comment]
Hi,

My HA is running on docker, can I add the modified file to custom_components?  Sure I have done this previously when there was a hotfix out a while ago before it was released on the HA update.  or will this method no longer work now?

Regards
Jack

**Comment by rikroe** [comment]
Yes, you can use the release of the custom component I have just created:
https://github.com/bimmerconnected/ha_custom_component/releases/tag/20210127.1

**Comment by jackshaw20** [comment]
Hi,
I installed from HACs last night and it did start to update both accounts, i have just updated to the latest release and now getting the same error back again.

Do I need to change line 123 to what was suggested?

Thanks 
Jack

**Comment by rikroe** [comment]
Yes, please change line 123 to the above.
The error will probably not occur always, it really depends on what is loaded at which point in time by HA, so you will have to monitor it for a while even if it works directly.

**Comment by jackshaw20** [comment]
changed the line and now seems to be updating both accounts again now, thanks :)     I will keep monitoring it as before it would sometimes work but then after a reboot one of the accounts would error.

**Comment by rikroe** [comment]
Cool! Please let me know if it really fixes the issue, then I'll create a PR with HA.

**Comment by rikroe** [comment]
Should be closed via home-assistant/core#47066
